app/services/internal-state.js

```diff
@@ -30,7 +32,23 @@ export default Service.extend({
   },
 
   getCurrentFolderID: function () {
-    return localStorage.currentFolderID;
+    let currentFolderID = localStorage.currentFolderID;
+    
+    // Skip checking falsey ID strings
+    if (typeof(currentFolderID) !== "undefined" && currentFolderID) {
+      this.store.findRecord('folder', currentFolderID).then(function(result) {
+        // folder exists - effectively a noop
+        // NOTE: after first lookup, this result is cached in the store
+      }).catch(function(error) {
+        //console.log(`Failed to fetch current folder (${currentFolderID}):`, error);
+        // Set currentFolderID to "undefined" and refresh the page
+        localStorage.removeItem("currentFolderID");
+        currentFolderID = undefined;
+        window.location.reload(true);
```

Comment:
I had not tried, but I had assumed that Ember is like most SPA frameworks in that it may only perform a navigation if the URL actually changes. I hadn't thought to try since we are only changing the values in localStorage, but I can give it a try this morning 😁

Reply:
Also, @bodom0015, fyi: https://www.npmjs.com/package/ember-window

Reply:
I attempted a hard-coded route, but sadly I was still left on the blank/blue screen:
```
 // Skip checking falsey ID strings
    let internalState = this;
    if (typeof(currentFolderID) !== 'undefined' && currentFolderID) {
      this.store.findRecord('folder', currentFolderID).then(function(result) {
        // folder exists - effectively a noop
        // NOTE: after first lookup, this result is cached in the store
      }).catch(function(error) {
        //console.log(`Failed to fetch current folder (${currentFolderID}):`, error);
        // Set currentFolderID to "undefined" and refresh the page
        localStorage.removeItem('currentFolderID');
        currentFolderID = undefined;
        internalState.router.transitionTo('/manage');
        //window.location.reload(true);
      });
    }
```

In the console, I see `ember-console.js:29 Error while processing route: manage.index`. Am I just doing it wrong? Maybe there is there a better / different method I should be using for the route transition?

Reply:
I have injected the service already, as you suggested:
```
  routerService: service('-routing'),
```

I think my issue is that I was expecting `transitionTo` to accept a URL path suffix, when it actually wants me to give it a route name.... do we know of a way to get the name of the current route? I am not even seeing where these are defined. For example, `login` => `/login`, but `manage.index` => `/manage`. 

I can access the current URL, but I am having trouble figuring out how to retrieve the actual route name to pass it to the router:
```
    // Skip checking falsey ID strings
    let internalState = this;
    if (typeof(currentFolderID) !== 'undefined' && currentFolderID) {
      this.store.findRecord('folder', currentFolderID).then(function(result) {
        // folder exists - effectively a noop
        // NOTE: after first lookup, this result is cached in the store
      }).catch(function(error) {
        //console.log(`Failed to fetch current folder (${currentFolderID}):`, error);
        // Set currentFolderID to "undefined" and refresh the page
        localStorage.removeItem('currentFolderID');
        currentFolderID = undefined;
        internalState.routerService.transitionTo('manage.index');
        //window.location.reload(true);
      });
    }
```

I should also mention that attempting to hard-code `'manage.index'` yielded a nasty-sounding stack trace that I was unable to decode without further research:
```
Uncaught Error: More context objects were passed than there are dynamic segments for the route: manage.index
```

Reply:
I had missed your comment about `ember-window` and took a look... it seems workable. I am, however, having trouble seeing the value-add right now since we are not yet trying to mock `window` for testing, or run outside of the browser, or add custom watchers for properties on `window`.

This was a quick fix for a bug before a release, and I figured we would want to avoid adding a new dependency at the last minute just before a release.

Reply:
> I had missed your comment about ember-window and took a look... it definitely seems like a better pattern long-term. I am, however, having trouble seeing the value-add right now since we are not yet trying to mock window for testing, or run outside of the browser, or add custom watchers for properties on window.This was a quick fix for a bug before a release, and I figured we would want to avoid adding a new dependency at the last minute just before a release.

I agree, this is nothing we need to support right now, just a heads up of what Ember dims "appropriate"

Reply:
Regarding the routes, there is a `router` object `app > router.js` that defines all the routes within this application @bodom0015

Reply:
In the case of manage:
```
    this.route('manage', function () {
        this.route('index', { path: '/'});
        this.route('view', { path: '/:image_id'});
    });
```

Reply:
The `window.location.reload(true);` solution is perfectly fine for now so we can ship the new version, @bodom0015. These are not show-stopper changes and your fix works as a charm, ok?